pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3124,13 +3176,30 @@ private PinotResourceManagerResponse enableInstance(String instanceName, boolean
         "Instance: " + instanceName + (enableInstance ? " enable" : " disable") + " failed, timeout");
   }
 
-  public RebalanceResult rebalanceTable(String tableNameWithType, Configuration rebalanceConfig)
+  /**
+   * Entry point for table Rebalacing.
+   * @param tableNameWithType
+   * @param rebalanceConfig
+   * @param helixZkManager
+   * @return RebalanceResult
+   * @throws TableNotFoundException
+   */
+  public static RebalanceResult rebalanceTable(String tableNameWithType, Configuration rebalanceConfig,
```

Comment:
Suggest not making it a static method. We should only use `ZkBasedTableRebalanceObserver` when it is triggered by the controller, and this method shouldn't be used by the tooling (`PinotTableRebalancer`)

Reply:
I wanted to unify all rebalance triggers in one place (PinotHelixResourceManager.rebalanceTable) to leverage code, however after our discussion, I've reverted this change. We decided that we want to track rebalance only for the controller (UI triggered/manual rebalance).